examples/crd_derive_schema.rs

```diff
@@ -0,0 +1,146 @@
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+// This example shows how the schema is generated, especially with nullable fields.
+//
+// TODO Write tests to verify the response from the server matches the commented behavior.
+//
+// References:
+// - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting
+// - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable
+
+#[derive(CustomResource, Serialize, Deserialize, Default, Debug, PartialEq, Clone, JsonSchema)]
+#[kube(
+    group = "clux.dev",
+    version = "v1",
+    kind = "Foo",
+    namespaced,
+    derive = "PartialEq",
+    derive = "Default"
+)]
+#[kube(apiextensions = "v1")]
+pub struct FooSpec {
+    // Required field
+    //
+    // There shouldn't be any ambiguity here.
+    required_field: String,
+
+    // Nullable without default.
+    //
+    // By skipping to serialize, the field won't be present in the object.
+    // If serialized as `null`, the object will have the field set to `null`.
+    //
+    // Deserializing works as expected either way. `None` if it's missing or `null`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    nullable: Option<String>,
+
+    // Non-nullable with default value.
+    //
+    // Serializing will work as expected because the field cannot be `None`.
+    //
+    // When deserializing a response from the server, the field should always be a string because
+    // the field is non-nullable and the server sets the value to the default specified in the schema.
+    //
+    // When deserializing some input, the default value will be set if missing.
+    // However, if `null` is specified, `serde` will panic.
+    // The server prunes `null` for non-nullable field since 1.20 and the default is applied.
+    // To match the server's behavior exactly, we can use a custom deserializer.
+    #[serde(default = "default_value")]
+    with_default: String,
+
+    // Nullable with default value.
+    //
+    // By skipping to serialize when `None`, the server will set the the default value specified in the schema.
+    // If serialized as `null`, the server will conserve it and the defaulting does not happen (since 1.20).
+    //
+    // When deserializing, the default value is used only when it's missing (`null` is `None`).
+    // This is consistent with how the server handles it since 1.20.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default = "default_nullable")]
+    nullable_with_default: Option<String>,
+}
```

Comment:
@clux 

> - `default` and `default_with` serde attrs should correspond with [crd defaulting rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting)
> - if `Option`al fields are [nullable](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable), do they require explicit serialization of the null value for k8s to null them out? If so, we ought to at least document that it's a bit scary to put `#[serde(skip_serializing_if = "Option::is_none")]` on them.

I added an example focused on CRD schema containing `FooSpec` with nullable fields and defaults. For each field, I added some comments describing how it behaves (or what I think how it behaves).

Creating `Foo` and checking the response should be enough to confirm these. Should I write something like `crd_api`?

Reply:
Awesome. That is just what I wanted; an example to showcase the new possibilities, and how it interacts.

> Should I write something like crd_api?
Yeah, please! Even if's just some simple `#[test]`cases below your new example.